### PR TITLE
XS✔ ◾ New rule JWTs - this weeks CTF.

### PR DESCRIPTION
**Pull request by Dhruv-0987**
**Tip: Use [SSW Rule Writer GPT](https://chat.openai.com/g/g-cOvrRzEnU-ssw-rules-writer) for help with writing rules 🤖**
>
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ New rule for this weeks CTF

> 2. What was changed?

✏️-

> 3. Did you do pair or mob programming (list names)?

✏️  I worked with @gordonbeeming

### CODE REVIEW

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+type: rule
+title: Do you securely implement and handle JWT Tokens?
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:3]: The T in JWT stands for token so "JWT Tokens" is redundant. I would suggest change to "Do you securely implement and handle Json Web Tokens (JWTs)?

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    url: https://www.ssw.com.au/people/dhruv-mathur/
+  - title: Gordon Beeming
+    url: https://www.ssw.com.au/people/gordon-beeming/
+created: 2024-03-06T13:42:55.753Z
+guid: fa645fa6-4054-448b-9b4b-d60a66422f1f
+---
+
+# Do you securely implement and handle JWT Tokens?
+
+JSON Web Tokens (JWTs) offer a compact and secure means of transmitting information between parties as JSON objects. Their ability to encode identity and claim information has made them popular in modern web development for authentication and authorization processes. However, handling JWTs securely is essential to prevent potential security breaches, such as token theft leading to unauthorized access.
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:16]: "claim" doesn't make sense here. Maybe "identity and other information" or just "information".

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+### Why JWTs Are Popular
+
+JWTs are favored for their:
+* **Compactness:** Making them easily transmissible via URL, POST parameters, or inside HTTP headers.
+* **Self-contained nature:** Carrying all necessary information, reducing the need for database lookups to authenticate requests.
+* **Versatility:** Being used in various scenarios, from Single Sign-On (SSO) to cross-service authentication in microservices architectures.
+
+### How JWT token are used
+
+JWTs token are used to implement authorised access for resouces (APIs) in an application, here is how a typical flow looks like:
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:34]: Extra word "token" here. Also inconsistent use of US vs correct spelling (you're using z above, which is the preferred approach for rules, and s here).

Also should be "here is how a typical flow looks" or "what a typical flow looks like".

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+* **User Authentication:** A user logs in to an application, and the server creates and issues a signed JWT token upon successful authentication.
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:36]: Redundant use of token. Also technically the user signs in to an authority, or IDP, or token issuer; not an application.

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+* **Token Storage:** The client application receives the JWT and stores it securely, typically in local storage or a secure cookie.
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:37]: Major problem here. Local storage is not secure.

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+* **Authorised access:** The client sends the JWT with each subsequent request to the server by putting it in the header of the request as a bearer token. The server when processing the request verifies the token's signature and the claims present, for granting access to the requested resource if the token is valid.
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:38]: Correct spelling here when it should be US spelling for Rules. Also I would replace "server" with "resource".

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+* **Token Expiry:** The server can set an expiry time for the token, after which the client must re-authenticate to obtain a new token. In a typical case a refresh token is also issued with the access token which can be used to obtain a new access token without the user having to re-authenticate.
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:39]: I would say the IDP/Authority/Issuer can set the expiry time for the token.

Also you have not explained here the difference between access tokens and refresh tokens.

Comment (matt-goldman) [reply rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:39]: ...to _silently_ obtain a new access token without the user having to re-authenticate _manually_ or _interactively_.

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+#### Asymmetric Algorithms (e.g., RSA, ECDSA)
+
+* **How it works:** Utilizes a public/private key pair; the JWT is signed with the private key and verified with the public key.
+* **Pros:** Increased security as the private key never needs to be shared; the public key can be openly distributed for token verification.
+* **Cons:** More computationally intensive, leading to slower performance.
+* **Best used when:** Multiple parties need to verify the token without sharing a secret, such as in distributed systems.
+
+### Best Practices for JWT Implementation
+
+1. **Keep Payloads Lean and Non-sensitive:** Avoid storing sensitive information in the JWT payload. If you must, ensure it's encrypted.
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:63]: I think you need to elaborate on this above. JWT payloads are not encrypted so you shouldn't put sensitive information in them. At all. This is just not really good advice, and even if it were, you would need to explain what you mean (i.e. encrypt the content of the claim, plus that's a whole other thing about how do you do that).

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+2. **Utilize Short Expiry Times:** This limits the window of opportunity for a token to be misused if compromised.
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:64]: I don't suggest changing this, but this is a comfort thing. Unless the expiry time is less than 8ms, this is not effective.

Better alternatives are either single use tokens (much more secure but much more overhead) or better yet DPoP which is gaining traction now.

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+3. **Secure Token Storage:** Implement secure mechanisms to store and transmit tokens, such as HTTPS and secure cookies.
```
Comment (matt-goldman) [inline rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:65]: TBH I thought this was what this whole rule was going to be about. This is what we need more advice on.

**Review by matt-goldman — CHANGES_REQUESTED** [review]
I have a few problems with this rule (in addition to the inline comments).

1. It talks about securely implementing JWTs. **Don't do this!!!** _Never_ role your own auth is the golden rule. IdentityServer is literally an add-on for ASP.NET Core for issuing and validating tokens.
2. There is no explanation of what a JWT is, e.g. base64 encoded, 3 sections (header, payload, signature) and what they are.
3. No mention of OIDC or where they are used.
4. No mention of how to actually use them securely in your applications. There is a mention of secure cookie, but no mention of DPoP or the BFF pattern. I also think there need to be links to relevant sources.
5. A fair few bits of bad advice in the rule. See comments.

Will try and catch you both this arvo for a chat about this.

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md (hunk 6 above)
Comment (lukecookssw) [reply rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md:38]: zubzequent

File: rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md (hunk 1 above)
Comment (Dhruv-0987) [reply rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md]: As per our conversation the rule was changed entirely

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+#### Token Expiry and Refresh Mechanism
+
+- JWTs typically have an expiration time set by the server. Once expired, the user must re-authenticate to obtain a new token.
+- Often, a refresh token mechanism is employed, where a longer-lived refresh token is issued alongside the JWT. The refresh token can be used to sliently obtain new access tokens without requiring the user to re-authenticate, enhancing the user experience while maintaining security.
+
+![Figure: JWT Authentication Flow](jwt-how-its-used.png)
+
+### Benefits of JWT in Stateless Authentication
+
+- **Scalability:** As the ExternalAuthProvider doesn't store session data, it can easily handle requests from a large number of users without a significant impact on performance.
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: I am a bit confused, didn't quite get his. I get why its wrong but not sure how

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    url: https://www.ssw.com.au/people/dhruv-mathur/
+  - title: Gordon Beeming
+    url: https://www.ssw.com.au/people/gordon-beeming/
+  - title: Matt Goldman
+    url: https://www.ssw.com.au/people/matt-goldman/
+created: 2024-03-06T13:42:55.753Z
+guid: fa645fa6-4054-448b-9b4b-d60a66422f1f
+---
+
+#Do you know how Modern Stateless Authentication works?
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: Done - added a new space.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+
+Modern stateless authentication, particularly through JSON Web Tokens (JWT), is a widely adopted method in web development for managing user sessions and securing applications. Unlike traditional session-based authentication, JWT allows for a more scalable and efficient approach by avoiding server-side storage of user session information.
+
+This new approach to authentication often utilizes the OpenID Connect (OIDC) authentication flow, which is an authentication layer built on top of OAuth 2.0. This protocol extends OAuth 2.0 by adding an ID token, which is a JSON Web Token (JWT) that contains information about the end-user
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md:20]: Done - reworded this line.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+A JWT is a base64-encoded JSON object that is digitally signed and optionally encrypted. It consists of three parts: a header, a payload, and a signature. The header contains metadata about the token, such as the type and the signing algorithm. The payload contains claims, which are statements about the user and additional data. The signature is used to verify the integrity of the token and to ensure that it has not been tampered with.
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md:22]: Done - added this.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+
+The process of stateless authentication usually involves trusting a ExternalAuthProvider outside  your application to authenticate users and then using a token issued by them to identify the user in your application. An ExternalAuthProvider can be a service like Azure AD, IdentityServer, or any other OAuth2 or OpenID Connect provider.
+
+{{insert video here...}}
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: Done - have inserted the link now.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+<!--endintro-->
+
+### The Process of Stateless Authentication with JWT
+
+#### User Authentication
+
+- The user submits their credentials (username and password) to the ExternalAuthProvider.
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: Done - replaced with the suggestion.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+- Upon successful authentication, the provider generates a JWT, which includes claims about the user and is signed with a secure key.
+- The JWT is then returned to the client application, usually via an HTTP response.
+
+#### Token Storage
+
+- The client application receives the JWT and stores it securely, often in the browser's local storage or an HttpOnly cookie to prevent access from JavaScript and reduce the risk of cross-site scripting (XSS) attacks.
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: Agreed - replaced it with the suggestion.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+- JWTs typically have an expiration time set by the server. Once expired, the user must re-authenticate to obtain a new token.
+- Often, a refresh token mechanism is employed, where a longer-lived refresh token is issued alongside the JWT. The refresh token can be used to sliently obtain new access tokens without requiring the user to re-authenticate, enhancing the user experience while maintaining security.
+
+![Figure: JWT Authentication Flow](jwt-how-its-used.png)
+
+### Benefits of JWT in Stateless Authentication
+
+- **Scalability:** As the ExternalAuthProvider doesn't store session data, it can easily handle requests from a large number of users without a significant impact on performance.
+- **Flexibility:** JWTs can be used across different domains and in various architectures, such as microservices, making them suitable for modern, distributed applications.
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: removed microservices.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+- The JWT is then returned to the client application, usually via an HTTP response.
+
+#### Token Storage
+
+- The client application receives the JWT and stores it securely, often in the browser's local storage or an HttpOnly cookie to prevent access from JavaScript and reduce the risk of cross-site scripting (XSS) attacks.
+
+#### Authorized Access
+
+- For subsequent requests to your resource avaibale at your application backend server, the client includes the JWT in the request headers as a Bearer token.
+- The server, upon receiving a request with a JWT, verifies the token's signature and the validity of its claims against an authority which in this case is the ExternalAuthProivder.
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: Done - replaced with the suggestion.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (matt-goldman) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: The application server doesn't need to manage sessions. It's true that the auth provider doesn't either, but the benefit in terms of scalability is that JWT auth is stateless, so the application doesn't have to manage sessions.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+    url: https://www.ssw.com.au/people/dhruv-mathur/
+  - title: Gordon Beeming
+    url: https://www.ssw.com.au/people/gordon-beeming/
+  - title: Matt Goldman
+    url: https://www.ssw.com.au/people/matt-goldman/
+created: 2024-03-06T13:42:55.753Z
+guid: fa645fa6-4054-448b-9b4b-d60a66422f1f
+---
+
+# Do you know how Modern Stateless Authentication works?
```
Comment (tiagov8) [inline rules/do-you-know-how-modern-stateless-auth-works/rule.md:16]: This is a duplicate of the title

- [ ] 1. Please remove

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+
+A JWT is a base64-encoded JSON object that is digitally signed and optionally encrypted. It consists of three parts: a header, a payload, and a signature. The header contains metadata about the token, such as the type and the signing algorithm. The payload contains claims, which are statements about the user and additional data. The signature is used to verify the integrity of the token and to ensure that it has not been tampered with and also to establish that the token has been issued by a trusted authority.
+
+The process of stateless authentication usually involves trusting a ExternalAuthProvider outside your application to authenticate users and then using a token issued by them to identify the user in your application. An ExternalAuthProvider can be a service like Azure AD, IdentityServer, or any other OAuth2 or OpenID Connect provider.
+
+<https://www.youtube.com/embed/UBUNrFtufWo>
+**Figure: Session vs Token Authentication (2min)**
+<!--endintro-->
```
Comment (tiagov8) [inline rules/do-you-know-how-modern-stateless-auth-works/rule.md:28]: Too much stuff in this intro... we usually have 1 or 2 paragraphs

- [ ] 1. Please move this divisor up

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+
+### The Process of Stateless Authentication with JWT
+
+#### User Authentication
+
+* The user authenticates to the ExternalAuthProvider.
```
Comment (tiagov8) [inline rules/do-you-know-how-modern-stateless-auth-works/rule.md:34]: - [ ] 1. Please follow https://rules.ssw.com.au/avoid-full-stops-in-bullet-point-lists in this list and all others

**Review by tiagov8 — CHANGES_REQUESTED** [review]

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+
+This new approach to authentication often utilizes the OpenID Connect (OIDC) authentication flow, which is an authentication layer built on top of OAuth 2.0. This protocol extends OAuth 2.0 by adding an ID token, which is a JSON Web Token (JWT) that contains information about the end-user
+
+A JWT is a base64-encoded JSON object that is digitally signed and optionally encrypted. It consists of three parts: a header, a payload, and a signature. The header contains metadata about the token, such as the type and the signing algorithm. The payload contains claims, which are statements about the user and additional data. The signature is used to verify the integrity of the token and to ensure that it has not been tampered with.
+
+The process of stateless authentication usually involves trusting a ExternalAuthProvider outside  your application to authenticate users and then using a token issued by them to identify the user in your application. An ExternalAuthProvider can be a service like Azure AD, IdentityServer, or any other OAuth2 or OpenID Connect provider.
```
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: I replaced ExternalAuthProvider with identity provider.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: replace ExternalAuthProvider with resource

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md (hunk 21 above)
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md:22]: Done - made the intro small and concise.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md (hunk 20 above)
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: >1. Please remove

✅Done - removed.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md (hunk 22 above)
Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: >1. Please follow https://rules.ssw.com.au/avoid-full-stops-in-bullet-point-lists in this list and all others

✅Done.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 24]
```diff
@@ -0,0 +1,67 @@
...
+    url: https://www.ssw.com.au/people/gordon-beeming/
+  - title: Matt Goldman
+    url: https://www.ssw.com.au/people/matt-goldman/
+created: 2024-03-06T13:42:55.753Z
+guid: fa645fa6-4054-448b-9b4b-d60a66422f1f
+---
+
+Modern stateless authentication, leveraging JSON web tokens (JWT), offers a scalable solution for managing user sessions without server-side storage. The OIDC framework, building on OAuth 2.0, standardizes interactions between authentication services and users. JWTs, containing a header, payload, and signature, ensure data integrity and authentication validity. Issued by trusted identity providers, these tokens facilitate user identification in applications, moving away from traditional session-based methods. This approach enhances scalability and user management, aligning with current web development trends.
+
+<https://www.youtube.com/embed/UBUNrFtufWo>
```
Comment (GordonBeeming) [inline rules/do-you-know-how-modern-stateless-auth-works/rule.md:18]: Is this how we embed videos? I thought there was a youtube: thing in there :)

Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: you are absolutely correct 😄 - I have added that now.

Comment (GordonBeeming) [inline rules/do-you-know-how-modern-stateless-auth-works/rule.md]: Let's also switch to the first video... I think it still fits with the contents of the rule

Comment (Dhruv-0987) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md]: Done.

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+    url: https://www.ssw.com.au/people/gordon-beeming/
+  - title: Matt Goldman
+    url: https://www.ssw.com.au/people/matt-goldman/
+created: 2024-03-06T13:42:55.753Z
+guid: fa645fa6-4054-448b-9b4b-d60a66422f1f
+---
+
+Modern stateless authentication, leveraging JSON web tokens (JWT), offers a scalable solution for managing user sessions without server-side storage. The OIDC framework, building on OAuth 2.0, standardizes interactions between authentication services and users. JWTs, containing a header, payload, and signature, ensure data integrity and authentication validity. Issued by trusted identity providers, these tokens facilitate user identification in applications, moving away from traditional session-based methods. This approach enhances scalability and user management, aligning with current web development trends.
+
+youtube: <https://www.youtube.com/embed/P2CPd9ynFLg>
```
Comment (GordonBeeming) [inline rules/do-you-know-how-modern-stateless-auth-works/rule.md:18]: I think it's meant to look like this

```suggestion
`youtube: https://www.youtube.com/embed/P2CPd9ynFLg`
```

File: rules/do-you-know-how-modern-stateless-auth-works/rule.md [hunk 26]
```diff
@@ -0,0 +1,67 @@
...
+  - title: Matt Goldman
+    url: https://www.ssw.com.au/people/matt-goldman/
+created: 2024-03-06T13:42:55.753Z
+guid: fa645fa6-4054-448b-9b4b-d60a66422f1f
+---
+
+Modern stateless authentication, leveraging JSON web tokens (JWT), offers a scalable solution for managing user sessions without server-side storage. The OIDC framework, building on OAuth 2.0, standardizes interactions between authentication services and users. JWTs, containing a header, payload, and signature, ensure data integrity and authentication validity. Issued by trusted identity providers, these tokens facilitate user identification in applications, moving away from traditional session-based methods. This approach enhances scalability and user management, aligning with current web development trends.
+
+`youtube: https://www.youtube.com/embed/P2CPd9ynFLg`
+**Figure: Why JWT is popular? (5min)**
```
Comment (GordonBeeming) [inline rules/do-you-know-how-modern-stateless-auth-works/rule.md:19]:
```suggestion
**Video: Why JWT is popular? (5 min)**
```

1. Videos have a Video prefix - https://www.ssw.com.au/rules/add-useful-and-concise-figure-captions/#tip-1-use-prefixes
2. Time length has a space by `min` - https://www.ssw.com.au/rules/right-format-to-write-videos-time-length/

Comment (matt-goldman) [reply rules/do-you-know-how-modern-stateless-auth-works/rule.md:19]: Also change to Why is JWT popular?

**Review by matt-goldman — APPROVED** [review]
LGTM 🚀

**Review by GordonBeeming — APPROVED** [review]

**Pull request merged**